static/styles/compose.css

```diff
@@ -736,3 +736,7 @@ a.compose_control_button.hide {
         flex: 1;
     }
 }
+
+.preview_mode {
+    visibility: hidden;
+}
```

Comment:
```
.preview_mode {
    .preview_mode_hidden {
        visibility: hidden;
    }
}
```

Let's do this here.

Reply:
Add `preview_mode_hidden` class to the icons we want to hide in `compsoe_control_buttons.hbs`. Then add `preview_mode` to class to the `#compsoe` div when preview mode is active via JS.

Reply:
or just use `$("#compsoe").toggleClass("preview_mode")` in click handlers.

Reply:
The first option is good via JS because while in preview mode user can close compose box or directly use the draft message, at that time hidden buttons stay hidden that's why when `clear_compose_box()` gets called we have to re-show hidden buttons, and now as we are calling it in `clear_compose_box()` we can't use `$("#compsoe").toggleClass("preview_mode") ` in it because buttons can go hidden even in write/edit mode.

Reply:
If I am missing something please elaborate : ).